File: 프로그래머스/2/12924. 숫자의 표현/숫자의 표현.cpp
#include <string>
#include <vector>
#include <bits/stdc++.h>

using namespace std;

int solution(int n) {
    int answer = 0;
    int start = 1;
    int end = 1;
    int sum = 1;
    
    while (start <= n)
    {
        if(sum == n)
            answer++;
        
        if(sum >= n)
        {
            sum -= start;
            start++;
        }
        else
        {
            end++;
            sum += end;
        }
    }
    
    
    return answer;
}

File: 프로그래머스/2/12924. 숫자의 표현/숫자의 표현.py
def solution(n): # 이분탐색?
    answer = 0
    for i in range(1, n+1):
        sm = 0
        for j in range(i, n+1):
            sm += j
            if sm == n:
                answer += 1
            elif sm > n:
                break
                
    return answer